content/developer/howtos/standalone_owl_application.rst

```diff
@@ -0,0 +1,163 @@
+the web client. One example in Odoo is the self-ordering application, that lets customers order
+food from their phone. In this chapter we will go into what's required to achieve something like this.
+
+Overview
+========
+
+To have a standalone Owl app, a few things are required:
+
+- a root component for the application
+- some setup code to start the services and mount the component
+- an assets bundle that contains the setup code
+- an QWeb view that calls the assets bundle
+- a controller that renders the view
+
+1. Root component
+=================
+
+To keep things simple, let's start with a very straightforward component that just renders
+"Hello, World!". This will let us know at a glance if our setup is working.
+
+.. code-block:: js
+
+    /** @odoo-module */
+    import { Component, xml } from "@odoo/owl";
+    class Root extends Component {
+        static template = xml`Hello, World!`;
+    }
+
+.. seealso::
+   Owl components :ref:`reference<frontend/components>`.
+
+2. Setting up the environment and mounting the application
+==========================================================
+
+An Owl application needs an environment, and the Odoo JavaScript framework needs that environment to
+contain the :ref:`services<frontend/services>`.
+Starting the services is also required to load the translations, which we need to do before mounting
+our Owl application, so that we can give Owl a working translation function.
+
+.. code-block:: js
+
+    /** @odoo-module */
+    import { Component, xml, whenReady, App } from "@odoo/owl";
+    import { makeEnv, startServices } from "@web/env";
+    import { templates } from "@web/core/templates";
+    import { _t } from "@web/core/l10n/translation";
+
+    class Root extends Component {
+        static template = xml`Hello, World!`;
+    }
+
+    whenReady().then(async () => {
+        const env = makeEnv();
+        await startServices(env);
+        const app = new App(Root, {
+            templates,
+            env: env,
+            dev: env.debug,
+            translateFn: _t,
+            translatableAttributes: ["data-tooltip"],
+        });
+        return app.mount(document.body);
+    });
+
+3. Creating an assets bundle containing our code
+================================================
+
+In the manifest of your module, create a new :ref:`assets bundle<reference/assets_bundle>`.
+It should include the `web._assets_core` bundle, which contains the Odoo JavaScript
+framework and the core libraries it needs (e.g. Owl and luxon), after which you can have a
+glob that adds all the files for your application in the bundle.
+
+.. code-block:: py
+
+    {
+        # ...
+        'assets': {
+            'your_module.assets_standalone_app': [
+                ('include', 'web._assets_helpers'),
+                'web/static/src/scss/pre_variables.scss',
+                'web/static/lib/bootstrap/scss/_variables.scss',
+                ('include', 'web._assets_bootstrap'),
```

Comment:
any idea/reason why the three instruction above aren't directly in assets_bootstrap? Would be nice if we could do 
```py
'your_module.assets_standalone_app': [
                ('include', 'web._assets_bootstrap'),
                ('include', 'web._assets_core'),
                'your_module/static/src/standalone_app/**/*',
]
```

Reply:
Not sure, I've added a point to the roadmap to look at scss bundles and clean things up, they're hard to use and it's easy to miss files. Since core depends on bootstrap I'd like to directly include `_assets_bootstrap` in `_assets_core` eventually.